### ISSUE TITLE
Bug: Map tile image gets saved with the wrong plus code

### ISSUE THREAD

**Issue by VilleKylmamaa** [PraxisMapper/PraxisGodotComponents#1]
Hello,

I would like to report a bug. Here are the reproduction steps:

1. Enter the OverheadView scene which fetches map tile images from the PraxisMapper server.
2. Quickly move around the map with arrow keys causing multiple simultaneous HTTP calls for map tiles.
3. Eventually the same map tile image might get saved for two different plus codes. You see this on the map as two duplicate images next to each other.

Map in the debugger with two duplicate map tiles on the middle row:
<img src="https://github.com/user-attachments/assets/ed8aa981-949a-440a-a490-e2e4acd31ea3" width="550">

Saved images on the device:
![image2](https://github.com/user-attachments/assets/bdc3c048-cac8-4e69-852e-b3c55509b011)

Once this has occurred and an image has been saved with the wrong plus code, there is no way to remedy the situation within the client. The only thing you can do is remove the images from the device so that they get fetched again.

It seems to be a race condition issue. One way to *alleviate* the issue would be to pass the `plusCode` with which the HTTP request was made to the `tile_called` function in `PraxisMapper/Controls/MapTile.gd`, and only save the image and draw the texture if the `plusCode` matches `currentTile`. Like so:

```gdscript
func tile_called(result, plusCode): # Added parameter plusCode
	print("tile called received body")
	request.response_data.disconnect(tile_called)
	
	if plusCode == currentTile: # Added check
		var image = Image.new()
		if (image.load_png_from_buffer(result) != OK):
			return "error2"
		
		var saved = image.save_png("user://MapTiles/" + plusCode + "-" + styleSet + ".png")
		if (saved != OK):
			print("Not saved: " + str(saved))
			
		var texture = ImageTexture.create_from_image(image)
		texRect.texture = texture

func LoadTile(plusCode):
    ...
    # New parameter passed like so:
    request.response_data.connect(tile_called.bind(plusCode))
    ...
```

However, this solution is not perfect as when the bug would get reproduced the map tile is instead initially left empty. It will get fetched again and then saved with the correct plus code, but only after moving the screen a bit. I am not very familiar with the codebase or GDScript yet so perhaps there is a better solution.

P.S. Thank you for the project, hoping to make something out of it!

**Comment by drakewill-CRL** [comment]
Thanks for the bug report!

I think the core cause for this was multiple requests to draw stuff could go off, and if they overlapped the requests would wait for the same frame and then grab the image. I have a fix uploaded for ScrollingCenteredMap that should fix the issue, though it will also stop redrawing the map while it draws.  Get the latest code and see if that fixes things for you.

**Comment by VilleKylmamaa** [comment]
I cloned this repo again and unfortunately the bug still seems to reproduce. Middle tile has been duplicated in the image, and the two tiles below it are in the wrong place. Wrong placement instead of duplication also happened before the recent changes.

<img src="https://github.com/user-attachments/assets/6fc4ab0f-d6df-4ca3-b445-a2de7a4ad586" width="550">

The gaps between the tiles are due to the newly cloned repo assuming this resolution (PraxisCore.gd):

```gdscript
#system config values. These are for Cell12 resolution images from detailed data.
var mapTileWidth = 320 
var mapTileHeight = 500
```

Previously I had changed `mapTileHeight` to 400. This is a separate issue, but how would you go about changing the `OverheadView` scene and/or PraxisMapper server to accommodate `mapTileHeight` 500?

**Comment by drakewill-CRL** [comment]
Oh, I misunderstood. I thought you were drawing the tiles on the device, not the server. I have been spending weeks on fixing that for client-side drawing. Your code is a good step towards fixing it, and it should work fine at walking speed on a mobile device. I'm going to include it while I work through this and figure out how to resolve this completely.

OverheadView and MapTile.gd were my first attempt at showing map tiles on the client, and that's painfully clear here. I should have done something to handle the case where a MapTile made a second request before the first one completed.

If you're using the server to draw tiles, it's easiest to update PraxisCore.mapTileHeight to 400 to fix the gaps in the OverheadView layout. The 500 value is for drawing high-resolution tiles on the client itself.  I'll make a note to do something about that discrepancy, even if its just "document the differences and when/why to use which".

**Comment by drakewill-CRL** [comment]
Check the latest code out now. In addition to your change, MapTile now cancels any existing request when it needs to be updated, and if it still somehow gets the wrong PlusCode, it asks again for the new one.

This should resolve out your issues both with "Wrong maptile is saved" and "blank ones dont update until you move again". Please check it out and let me know.

**Comment by VilleKylmamaa** [comment]
Great work! The issue is resolved, and the would-be blank ones indeed update without having to move.
